app/src/main/java/com/rosario/covid19/data/CustomDataDeserializer.kt

```diff
@@ -0,0 +1,39 @@
+package com.rosario.covid19.data
+
+import com.google.gson.JsonDeserializationContext
+import com.google.gson.JsonDeserializer
+import com.google.gson.JsonElement
+import com.google.gson.JsonObject
+import com.rosario.covid19.data.model.DataResponse
+import com.rosario.covid19.data.model.Report
+import com.rosario.covid19.util.*
+import java.lang.reflect.Type
+
+
+class CustomDataDeserializer : JsonDeserializer<DataResponse> {
+    override fun deserialize(
+        json: JsonElement?,
+        typeOfT: Type?,
+        context: JsonDeserializationContext?
+    ): DataResponse {
+        val jsonObject = json!!.asJsonObject
+        val jsonData = jsonObject.get(DATA)
+        return if (jsonData is JsonObject) {
+            val report = Report()
+            report.date = jsonData.get(DATE).asString
```

Comment:
here you could use the gson to get the object

Reply:
this change has already been made